internal/zinc-persist/src/main/scala/sbt/internal/inc/cached/CacheAwareStore.scala

```diff
@@ -8,23 +8,28 @@
 package sbt.internal.inc.cached
 
 import java.io.File
+import java.util.Optional
 
-import sbt.internal.inc.AnalysisStore
-import xsbti.compile.{ MiniSetup, CompileAnalysis }
+import sbt.internal.inc.ConcreteAnalysisContents
+import xsbti.compile.{ AnalysisContents, AnalysisStore }
 
 case class CacheAwareStore(localStore: AnalysisStore,
                            cacheProvider: CacheProvider,
                            projectLocation: File)
     extends AnalysisStore {
-  override def set(analysis: CompileAnalysis, setup: MiniSetup): Unit = {
-    localStore.set(analysis, setup)
-  }
+  override def set(analysisFile: AnalysisContents): Unit =
+    localStore.set(analysisFile)
 
-  override def get(): Option[(CompileAnalysis, MiniSetup)] = {
-    val previous = localStore.get()
+  final val Empty = Optional.empty[AnalysisContents]
+  override def get: Optional[AnalysisContents] = {
+    import sbt.internal.inc.JavaInterfaceUtil.PimpOptional
+    val previous = localStore.get().toOption.map(f => (f.getAnalysis, f.getMiniSetup))
     val cache = cacheProvider.findCache(previous)
     val cachedResult = cache.flatMap(_.loadCache(projectLocation))
-
-    cachedResult.orElse(previous)
+    val optResult = cachedResult.orElse(previous)
+    optResult match {
+      case Some((analysis, setup)) => Optional.of(ConcreteAnalysisContents(analysis, setup))
+      case None                    => Empty
+    }
```

Comment:
This is just some internal data structures that @gkossakowski added but that they don't really belong to this project (though they are good for testing). What this does is that it enables to do more "custom" behaviour as a build tool would do, have a look at `ProjectRebase`.
  
These utils kept being private on purpose, and have not been removed in this PR to keep the tests that use the mappers under the hood.
  
The way to good is the public API, that is `FileAnalysisStore` + mappers.